tests/source/issue-4499.rs

```diff
@@ -0,0 +1,180 @@
+// Different "rustfmt::skip" (not)formatting tests
+
```

Comment:
Thank you for the thoroughness of the test suite! I do have a couple requested changes though:

* please try to keep issue-specific test files specific to those reported issues. it's always great to add more tests, but those don't need to be stuffed in to the same file
* this file is a little jarring given the size, and the excessive misformatting. i believe it will be much more readable and maintainable to break this into smaller chunks, so lets split these up in multiple files (feel free to create a new directory, e.g. `skip` to place them if nothing along those lines exists yet)
* lets refrain from having using comment text in these test cases which describes behavior that is fixed. it's okay for a small snippet from the original issue (though still free to change the words of the comment, as the actual words don't matter), but we don't want to repeat wording that will be very confusing to someone looking at this down the road
* lets avoid having duplicative attributes everywhere, again doing so makes it more difficult to review and eyeball since one has to try to manually match up the 5th instance of the same attribute between two files. The actual attribute doesn't really matter for our formatting purposes (doesn't have to be real/valid) so always prefer distinctness
* i'd advise against having _so_ much indentation, unless you feel it's really necessary to visually tell that the content is indeed skipped. I can certainly understand one or two levels, but some of these look a bit excessive for a test that's basically targeting idempotency between source and target (albeit with a couple exceptions)

Reply:
>    please try to keep issue-specific test files specific to those reported issues. it's always great to add more tests, but those don't need to be stuffed in to the same file .... this file is a little jarring given the size, and the excessive misformatting. i believe it will be much more readable and maintainable to break this into smaller chunks .....

Split the test file into several files in the skip folder (that already exited in target).  Did keep the "issue-4499" prefix for the file names.

>    lets refrain from having using comment text in these test cases which describes behavior that is fixed .... lets avoid having duplicative attributes everywhere ....

Change.  Hopefully it is o.k. now.  It seems that I prepared a draft of the test cases and somehow forgot that it is a draft ...

>   i'd advise against having so much indentation ...

Changed all source indentations to 3 tabs.  The reason is that 1 or 2 tabs may be the output of formatted code.  I hope this is o.k.

Reply:
> Did keep the "issue-4499" prefix for the file names.

Please do not do this. The point of having issue specific file names is to be able to quickly and succinctly tie back a test case to a reported issue. It is always fine to add additional test cases not directly related to a reported issue, but don't conflate the additional test cases with the reported issue.

Reply:
>> Did keep the "issue-4499" prefix for the file names.

> Please do not do this.

O.k.  Removed the "issue-4499-" prefix from the test files names.

Reply:
I won't block the PR as this is getting into bikeshedding territory, but you didn't need to go from one extreme to the other :smile:

Having a specific file for a reported issue to be used for system+idempotence tests is not mutually exclusive with adding other test files.

The way we handle this is pretty straightforward in my mind, but please let me know if this is unclear and/or if you have any questions. When creating specific test case content to cover a reported issue, there's a few options depending on the report/preference:

* Create a single source/target file pair that contains the MCVE (repro case, minimal complete and verifiable example) provided in the original issue, and name the file to match with a dash or underscore (e.g.`tests/{source,target}/issue-5004.rs`)
* If the originally provided MCVE requires multiple files to reproduce (e.g. module loading scenarios) then create a directory named after the issue and add the files within (issue name does not need to be tacked on to the file names, e.g. `tests/{source,target}/issue-5000/{foo.rs,bar.rs}`)
* Add the MCVE snippet to an existing test file, and use inline comments and/or function names to add the tie back to the reported issue, e.g. https://github.com/rust-lang/rustfmt/blob/87e960271b4036d37f73490bec8cba67f93b94ed/tests/target/expr.rs#L189-L195

If during the process of resolving a reported issue you discover that there's other scenarios we should have test cases for then that's great! However, you do not need to, and should not, tie those additional scenarios to the reported issue just because they were conjured while thinking about or working on a reported issue. The above naming conventions and test file locations _only_ apply for the reported MCVEs, and other tests can be added elsewhere